doc/modules/cassandra/pages/tools/hash_password.adoc

```diff
@@ -0,0 +1,26 @@
+= Hash password
+
+The `hash_password` tool is used to get the jBcrypt hash of a password. This hash 
+can be used in CREATE/ALTER ROLE/USER statements for improved security.
+
+This feature can be useful if we want to make sure no intermediate system, logging or 
+any other possible plain text password leak can happen.
+
+== Usage
+
+....
+One of the options --environment-var, --plain or --input must be used.
```

Comment:
This seems to be a capture of the output of calling the tool without arguments, where this first line is the error message and it should probably be removed from the header. Maybe it could go at the end of the file, given that it's useful info and it's not included in the output of `tools/bin/hash_password -h`?

Reply:
Correct, this was intentional bc I found it useful but I can move it around.

Reply:
I think it reads better with the message about mandatory options at the end. I find more natural to first read the description, then the `hash_password [options]` syntax, then the list of options, and finally the rules about the previously introduced options. Otherwise, the `One of the options --environment-var, --plain or --input must be used.` message mentions options that haven't been yet introduced. But that's a nit.

Also, the `Hashes a plain text password and prints the hashed password` part seems a bit reiterative with the description section right above. Maybe we could use the format used for sstable tools instead of the command output, this way:
```
= Hash password

The `hash_password` tool is used to get the jBcrypt hash of a password. This hash 
can be used in CREATE/ALTER ROLE/USER statements for improved security.

This feature can be useful if we want to make sure no intermediate system, logging or 
any other possible plain text password leak can happen.

== Usage

hash_password <options>

[cols=",",]
|===

|-h,--help |Displays help message

|-e,--environment-var <arg> |Use value of the specified environment
variable as the password

|-i,--input <arg> |Input is a file (or - for stdin) to read the
password from. Make sure that the whole input including newlines is
considered. For example, the shell command `echo -n foobar \| hash_password
-i -` will work as intended and just hash 'foobar'.

|-p,--plain <arg> |Argument is the plain text password

|-r,--logrounds <arg> |Number of hash rounds (default: 10).
|===

One of the options --environment-var, --plain or --input must be used.
```
wdyt?

Reply:
I think we're splitting hairs. I still prefer it at the beginning but we're on the personal taste area maybe so I am happy to move it around.

The columnar help text is the same. In the docs we have both the tool help copy-paste based approach and the columnar approach. The text one reads better to my eyes but again, personal taste imo.

Happy to put it that way as you already did the work.